cli/npm/managed/resolvers/local.rs

```diff
@@ -609,16 +609,79 @@ async fn sync_resolution_with_fs(
     }
   }
 
-  // 4. Create all the top level packages in the node_modules folder, which are symlinks.
-  //
-  // Symlink node_modules/<package_name> to
-  // node_modules/.deno/<package_id>/node_modules/<package_name>
-  let mut found_names = HashSet::new();
-  let mut ids = snapshot.top_level_packages().collect::<Vec<_>>();
+  let mut found_names: HashMap<&String, &PackageNv> = HashMap::new();
+
+  // 4. Create symlinks for package json dependencies
+  {
+    for remote in pkg_json_deps_provider.remote_pkgs() {
+      let remote_id = snapshot
+        .resolve_best_package_id(&remote.req.name, &remote.req.version_req)
```

Comment:
Is there a better way to resolve the version req to a package id?

Reply:
This is the way to do it. I'm not sure why the tests are panicking on this line though.

Reply:
Ah if you have
```
deno cache npm:@types/node
```

but your package.json contains a dep on `@denotest/add`, then the req will still be there, but `@denotest/add` won't be in the snapshot

Reply:
Ah, yeah if it doesn't match a package in the package.json then it won't auto-install the package.json dependencies.